contracts/LRTDepositPool.sol

```diff
@@ -164,13 +177,95 @@ contract LRTDepositPool is ILRTDepositPool, LRTConfigRoleChecker, PausableUpgrad
+    // its respective underlying EigenLayer strategy.
+    /// @dev Will emit the `Withdrawal` event from EigenLayer's `DelegationManager` contract
+    /// which is needed for the `claimWithdrawal` call.
+    /// @param asset address of the liquid staking token (LST) being requested. eg OETH. Can not be WETH.
+    /// @param assetAmount the amount of LSTs to withdraw.
+    /// @param maxPrimeETH the maximum amount of primeETH tokens that can be burned.
+    /// @return primeETHAmount the amount of primeETH tokens that were burned.
+    function requestWithdrawal(
+        address asset,
+        uint256 assetAmount,
+        uint256 maxPrimeETH
+    )
+        external
+        whenNotPaused
+        nonReentrant
+        returns (uint256 primeETHAmount)
+    {
+        if (assetAmount == 0) {
+            revert ZeroAmount();
+        }
+
+        // Can only withdraw the supported asset
+        if (asset != WITHDRAW_ASSET) {
+            revert NotWithdrawAsset();
+        }
+
+        // Convert asset amount to primeETH amount.
+        // Using mint function here even though it's a withdrawal as it does the same calculation
+        primeETHAmount = getMintAmount(asset, assetAmount);
+
+        // Check the primeETH amount to be burned is within the maximum allowed
+        if (primeETHAmount > maxPrimeETH) {
+            revert MaxBurnAmount();
+        }
+
+        // burn primeETH from the staker
+        address primeETH = lrtConfig.primeETH();
+        IPrimeETH(primeETH).burnFrom(msg.sender, primeETHAmount);
+
+        // Get the NodeDelegator contract to request the withdrawal from
+        address nodeDelegator = nodeDelegatorQueue[LST_NDC_INDEX];
+
+        // Calculate the strategy shares for the requested assets
+        address strategyAddress = lrtConfig.assetStrategy(asset);
+        uint256 strategyShares = IStrategy(strategyAddress).underlyingToShares(assetAmount);
+
+        INodeDelegator(nodeDelegator).requestWithdrawal(strategyAddress, strategyShares, msg.sender);
+
+        emit WithdrawalRequested(msg.sender, asset, strategyAddress, primeETHAmount, assetAmount, strategyShares);
+    }
+
+    /// @notice PrimeETH staker claims the withdrawal of their previously requested liquid staking token (LST).
+    /// Must wait `minWithdrawalDelayBlocks` on EigenLayer's `DelegationManager` contract
+    /// before claiming the withdrawal.
+    /// This is currently set to 50,400 blocks (7 days) on mainnet. 10 blocks on Holesky.
+    /// @dev The asset is validated against the withdrawal strategy in EigenLayer's `StrategyBase`.
+    /// @return asset the address of the LST that was withdrawn
+    /// @return assets the amount of LSTs received from the withdrawal
+    function claimWithdrawal(IDelegationManager.Withdrawal calldata withdrawal)
+        external
+        whenNotPaused
+        nonReentrant
+        returns (address asset, uint256 assets)
```

Comment:
Let's also add a check on a supported asset to claim withdraw.

Reply:
The call to `claimWithdrawal` on the `NodeDelegator` will prevent an invalid asset being included in the claim as the hash of the withdrawal need to match the previously requested withdrawal by the user.

There is also no asset in the `withdrawal` param of `claimWithdrawal`. It has an array of `strategies` that was constructed by the `requestWithdrawal` function on the `NodeDelegator`.

Reply:
One possible scenarios is an asset could have been valid when `requestWithdrawal` was called but then changed to be invalid by the time `claimWithdrawal` is called.
I think we can use `requestWithdrawal` as the cutoff for validating an asset. If someone got a request in then they should be able to proceed with the claim.